compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -1311,7 +1311,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 let InferOk { obligations, .. } = self
                     .infcx
                     .at(&obligation.cause, obligation.param_env)
-                    .eq(DefineOpaqueTypes::No, target, new_struct)
+                    .eq(DefineOpaqueTypes::Yes, target, new_struct)
```

Comment:
Similar, doesn't this allow us to unsize `Adt<[impl Sized; 2]>` to `Adt<[Ty]>` (for example)